### PR TITLE
DOC: Enable nitpicky in doc build

### LINKED ISSUE #812: Enable nitpicky mode in docs building
Currently the two following lines are commented out in conf.py for the docs:

```
# nitpicky = True
# nitpick_ignore = []
```

We should ideally uncomment these and fix any warnings that come up and make sure we build the docs such that it will error if any warnings are emitted - this ensures that any intersphinx links are correct and is much easier to do earlier on in a project than later on - the more we wait to enable it the more of a pain it will be to fix all the warnings in one go 😄 

[🐱](https://jira.stsci.edu/browse/JDAT-1686)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is fix #812 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: docs/imviz/notebook.rst [hunk 1]
```diff
@@ -10,3 +10,9 @@ To run Imviz in a notebook::
     imviz = Imviz()
     imviz.app
 
+Imviz also provides programmatic access to its viewers using
+`Astrowidgets <https://astrowidgets.readthedocs.io/en/latest/>`_ API.
+For example::
+
+    viewer = imviz.get_viewer('viewer-1')
```
Comment (pllim) [inline docs/imviz/notebook.rst:17]: This will be true after #854 is merged.

File: docs/mosviz/plugins.rst [hunk 2]
```diff
@@ -67,6 +67,5 @@ A slit can be added to the image viewer by opening the Slit Overlay plugin and c
 The :guilabel:`Remove` button can be used to remove a slit once it has been applied to the image viewer.
 
 In order to plot a slit onto the image viewer, we need WCS information from an image and slit position from a 2D spectrum.
-WCS information is taken from the `meta` attribute of the :class:`~astropy.nddata.CCDData` object representing the data in the
```
Comment (pllim) [inline docs/mosviz/plugins.rst:70]: This is no longer true after #850 is merged, so I removed it here.

File: docs/nitpick-exceptions [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# https://github.com/mariobuikhuizen/ipyvuetify/issues/180
```
Comment (pllim) [inline docs/nitpick-exceptions:1]: xref mariobuikhuizen/ipyvuetify#180

File: docs/nitpick-exceptions [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# https://github.com/mariobuikhuizen/ipyvuetify/issues/180
+py:class ipyvuetify.VuetifyTemplate.VuetifyTemplate
+
+# https://github.com/glue-viz/glue-jupyter/issues/258
```
Comment (pllim) [inline docs/nitpick-exceptions:4]: xref glue-viz/glue-jupyter#258

File: docs/specviz/import_data.rst [hunk 5]
```diff
@@ -40,16 +40,16 @@ Here, users can select the loaded data set to be visualized in the viewer.
 Loading data via the API
 ------------------------
 Alternatively, if users are working in a coding environment like a Jupyter notebook, they have access to the :class:`~jdaviz.configs.specviz.helper.Specviz` helper class API. Using this API, users can load data into the application through code.
-Below is an example of importing the :class:`~jdaviz.configs.specviz.helper.Specviz` helper class, creating a :class:`~specutils.Spectrum1D` object from a data file via the :func:`~specutils.Spectrum1D.read` method::
+Below is an example of importing the :class:`~jdaviz.configs.specviz.helper.Specviz` helper class, creating a :class:`~specutils.Spectrum1D` object from a data file via the `specutils.Spectrum1D.read` method::
```
Comment (pllim) [inline docs/specviz/import_data.rst:43]: @rosteen pointed out that `specutils.Spectrum1D.read` is listed as "attribute" upstream even though it is a method. 🤷

File: jdaviz/app.py [hunk 6]
```diff
@@ -197,37 +197,25 @@ def __init__(self, configuration=None, *args, **kwargs):
...
     def hub(self):
         """
-        Reference to the stored application handler hub instance.
-
-        Returns
-        -------
-        `glue.core.Hub`
-            The internal ``Hub`` instance for the application.
+        Reference to the stored application handler `~glue.core.hub.Hub` instance
+        for the application.
```
Comment (pllim) [inline jdaviz/app.py:201]: Note: `@property` do not really need a "Returns" section. They must return something by definition.

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 7]
```diff
@@ -303,16 +303,17 @@ def add_data(self, data, color=None, alpha=None, **layer_state):
 
         Parameters
         ----------
-        spectrum : :class:`glue.core.Data`
+        spectrum : :class:`glue.core.data.Data`
             Data object with the spectrum.
-        color: color specification
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:308]: Unless someone can tell me exactly what Python data types "color specification" refers to, we should just leave it blank.

**Review by rosteen — APPROVED** [review]
LGTM!

**Pull request merged**